Plugins/ASEP/Get-ASEP.ps1

```diff
@@ -0,0 +1,176 @@
+<#
+
+.SYNOPSIS
+    Plugin-Name: Get-ASEP.ps1
+    
+.Description
+    This plugin gathers Auto Start Execution Points from one or several hosts
+    using the autorunsc Sysinternals tool. This plugin runs autoruns with the
+    following options on each host:
+
+    autorunsc64.exe -accepteula -a * -h -nobanner -v -s -t -c
+
+    The options that are inlcuded in this execution are:
+
+    -a             Specifies the ASEP type to collect (See below)
+    -h             Collects several hashes of each ASEP
+    -v             Queries VirusTotal based on hash only
+    -s             Verifies signatures of all ASEPs
+    -t             Standardizes time in UTC format
+    -c             Outputs in a CSV format
+    -nobanner      Does not inlcude Autoruns banner
+    -accepteula    Does not prompt for EULA acceptance
+
+    This plugin opts for collection of all ASEPs available including:
+
+    Boot execute, codecs, appinit DLLs, Explorer addons, sidebar gadgets, image
+    highjacks, Internet Explorer addons, known DLLs, logon startups, WMI entries,
+    Winsock protocol and network providers, Office addins, printer monitor DLLs,
+    LSA security providers, Autostart services and non-disabled drivers, scheduled
+    tasks, winlogon entries
+
+    Note: If you are interested in running a different version of this command, you 
+    can simply edit the Invoke-Command entry below and adjust as needed.
+
+    Note: If you are using custom directory paths, you will need to edit the EXE 
+    checks in the script so that it does not error and exit. Best practice, use the
+    default paths provided by Power-Response.
+
+.EXAMPLE
+    Get-ASEP.ps1 -ComputerName Test-PC
+
+    PowerResponse Execution
+    Set ComputerName Test-PC
+    Run
+
+.NOTES
+    Author: 5yn@x
+    Date Created: 12/29/2018
+    Twitter: @5ynax
+    
+    Last Modified By:
+    Last Modified Date:
+    Twitter:
+  
+#>
+
+param (
+
+    [Parameter(Mandatory=$true,Position=0)]
+    [string[]]$ComputerName
+
+    )
+
+process{
+
+    #Get the root Power-Response directory - Assumes that the plugin is located in the plugins/ASEP directory
+    $PRRoot = (Get-Item $PSScriptRoot).parent.parent.FullName
+
+    #Autorunsc executable locations
+    $Autorunsc64 = "$PRRoot\Bin\autorunsc64.exe"
+    $Autorunsc32 = "$PRRoot\Bin\autorunsc.exe"
+
+    #Verify binaries exist in Bin
+    $64bitTestPath = Get-Item -Path $Autorunsc64 -ErrorAction SilentlyContinue
+    $32bitTestPath = Get-Item -Path $Autorunsc32 -ErrorAction SilentlyContinue
+
+    if (-not $64bitTestPath) {
```

Comment:
For this error handling section, may want to use `throw` statements rather than `Write-Error`.

Difference between the statements is `throw` sends a script terminating error, while `Write-Error` is non-terminating. The framework is set up nicely to handle terminating errors, less so for non-terminating.

Would look like:
```
if (-not $64bitTestPath) {
    throw "Autorunsc64.exe not detected in Bin. Place 64bit executable in Bin directory and try again."
} else {
...
```